pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -658,49 +651,45 @@ public PinotResourceManagerResponse updateServerTenant(Tenant serverTenant) {
     boolean isCurrentTenantColocated =
         (allServingServers.size() < taggedOfflineServers.size() + taggedRealtimeServers.size());
     if (isCurrentTenantColocated != serverTenant.isCoLocated()) {
-      res.status = ResponseStatus.failure;
-      res.message = "Not support different colocated type request for update request: " + serverTenant;
-      LOGGER.error(res.message);
-      return res;
+      String message = "Not support different colocated type request for update request: " + serverTenant;
```

Comment:
The message does not read correctly. Would be great if it can be made gramatically correct.